proto/cosmos/crisis/v1/tx.proto

```diff
@@ -0,0 +1,97 @@
+syntax = "proto3";
+package cosmos.crisis.v1;
+
+import "cosmos/msg/v1/msg.proto";
+
+// Msg defines the crisis Msg service.
+service Msg {
+  option (cosmos.msg.v1.service) = true;
+
+  // AuthorizeCircuitBreaker allows a super-admin to grant (or revoke) another
+  // account's circuit breaker permissions.
+  rpc AuthorizeCircuitBreaker(MsgAuthorizeCircuitBreaker) returns (MsgAuthorizeCircuitBreakerResponse);
+
+  // TripCircuitBreaker pauses processing of Msg's in the state machine.
+  rpc TripCircuitBreaker(MsgTripCircuitBreaker) returns (MsgTripCircuitBreakerResponse);
+
+  // ResetCircuitBreaker resumes processing of Msg's in the state machine that
+  // have been been paused using TripCircuitBreaker.
+  rpc ResetCircuitBreaker(MsgResetCircuitBreaker) returns (MsgResetCircuitBreakerResponse);
+}
+
+message MsgAuthorizeCircuitBreaker {
+  option (cosmos.msg.v1.signer) = "granter";
+
+  string granter = 1;
+  string grantee = 2;
+  CircuitBreakerPermissions permissions = 3;
+}
+
+message CircuitBreakerPermissions {
+  // level is the level of permissions granted to this account.
+  Level level = 1;
+
+  // limit_type_urls is used with LEVEL_SOME_MSGS to limit the lists of Msg type
+  // URLs that the account can pause. It is an error to use limit_type_urls with
+  // a level other than LEVEL_SOME_MSGS.
+  repeated string limit_type_urls = 2;
+
+  enum Level {
+    // LEVEL_NONE_UNSPECIFIED indicates that the account will have no circuit
+    // breaker permissions.
+    LEVEL_NONE_UNSPECIFIED = 0;
+
+    // LEVEL_SOME_MSGS indicates that the account will have permission to
+    // trip or reset the circuit breaker for some Msg type URLs. If this level
+    // is chosen, a non-empty list of Msg type URLs must be provided in
+    // limit_type_urls.
+    LEVEL_SOME_MSGS = 1;
+
+    // LEVEL_ALL_MSGS indicates that the account can trip or reset the circuit
+    // breaker for Msg's of all type URLs.
+    LEVEL_ALL_MSGS = 2;
+
+    // LEVEL_SUPER_ADMIN indicates that the account can take all circuit breaker
+    // actions and can grant permissions to other accounts.
+    LEVEL_SUPER_ADMIN = 3;
+  }
+}
+
+message MsgAuthorizeCircuitBreakerResponse {}
+
+message MsgDeauthorizeCircuitBreaker {
```

Comment:
How is `MsgDeauthorizeCircuitBreaker` expected to work?

Reply:
Oh, it's not needed. I forgot to delete it. You just need to use `MsgAuthorizeCircuitBreaker` with level NONE